server/server.go

```diff
@@ -1711,14 +1753,35 @@ func (server *BgpServer) deleteNeighbor(c *config.Neighbor, code, subcode uint8)
 	return nil
 }
 
+func (s *BgpServer) DeletePeerGroup(c *config.PeerGroup) error {
+	return s.mgmtOperation(func() error {
```

Comment:
I think we should not delete a peer group when some neighbor is using it as `(*BgpServer).DeleteVrf()` doing.